### PR TITLE
Hook for custom middleware

### PR DESCRIPTION
**Pull request by sresant**
Relates to #694.

### CODE REVIEW

File: docs/guides/react-server-cli.md [hunk 1]
```diff
@@ -114,9 +114,22 @@ export default (webpackConfig) => {
 
 In the `.reactserverrc` file add an option for `webpack-config` that points to that function file and when React Server is setting up Webpack it will call your function with the result of the built in Webpack options, allowing you to make any modifications needed.
 
+### Use Custom Express Middleware
```
Comment (sresant) [inline docs/guides/react-server-cli.md]: @gigabo i am the worst at documentation so let me know what you think

**Comment by gigabo** [comment]
Hi @doug-wade!  It looks like @sresant has actually accounted for that.  The _default_ behavior remains to apply the various other middlewares (compression, etc) in a certain order and then apply the React Server middleware.  But if the new _hook_ is used the caller becomes responsible for applying _all_ middleware, _including_ compression, etc _and_ React Server's, in whatever order is desired.

File: docs/guides/react-server-cli.md [hunk 2]
```diff
@@ -114,9 +114,22 @@ export default (webpackConfig) => {
...
+	server.use(bodyParser.urlencoded({ extended: false }));
+	server.use(bodyParser.json());
+	server.use(session(options));
+	rsMiddleware(); // Must be called once or server will not start
+}
+```
 
 
+In the `.reactserverrc` file add an option for `custom-middleware-path` that points to that function file and when React Server is setting up the server it will call your function for setup rather then the default middlewares mentioned above. On the command line the option is `--custom-middleware-path`.
+
```
Comment (gigabo) [inline docs/guides/react-server-cli.md]: I think we've generally used the camelCase variants in the config file?  Like `customMiddlewarePath`?

Not sure if either will work, or even if I'm remembering right. ;)

Comment (sresant) [inline docs/guides/react-server-cli.md]: You are correct, i was just tired when writing up the documentation good catch!

File: docs/guides/react-server-cli.md [hunk 3]
```diff
@@ -114,9 +114,22 @@ export default (webpackConfig) => {
...
+### Use Custom Express Middleware
+Currently the default Express Middlewares used are compression, body-parser, cookie-parser. If you need to setup custom express middleware you can do it with a setup function.
 
+```javascript
+export default (server, reactServerMiddleware) => {
+	server.use(compression());
+	server.use(bodyParser.urlencoded({ extended: false }));
+	server.use(bodyParser.json());
+	server.use(session(options));
+	rsMiddleware(); // Must be called once or server will not start
```
Comment (sresant) [inline docs/guides/react-server-cli.md]: Good catch!

**Pull request merged**